specs/grafana_event_handling_in_monitoring_integration.adoc

```diff
@@ -0,0 +1,342 @@
+
+None
+
+==== Tendrl/node_agent impact:
+
+Node-agent must be modified to insert the events received by Monitoring
+Integration in etcd under alerting/alerts/ instead of messages/events/.
+
+Monitoring integration adds scope ( global, cluster specific or node specific)
+of alert in the message metadata, this scope should be used by node-agent
+to place the event in the proper path in etcd.
+
+==== Sds integration impact:
+
+None
+
+==== Tendrl/performance-monitoring impact:
+
+A  new component "Monitoring Integration" will be introduced to
+handle events generated by grafana.
+
+In this component:
+
+Create a new class called AlertHandler and run this class as separate gevent.
+AlertHandler will receive the alert event from socket and convert that alert
+as dictionary based on current alert structure and pass the alert as a message
+to the node-agent socket. (more details in implementation section)
+
+```
+To fetch details regarding particular alert from grafana:
+   http://{ip}:3000/api/alerts/{id}
+```
+
+==== Tendrl/alerting impact:
+
+Tendrl alerting system will be renamed as Notifier.
+Its main functionality would be to fetch the alert events and
+notify the respective user.
+
+=== Security impact:
+ 
+None
+ 
+=== Other end user impact:
+
+Users will be able to set alerts in Grafana itself and get notified for the same.
+
+=== Performance impact:
+
+None
+
+=== Other deployer impact:
+
+None
+
+=== Developer impact:
+
+Need to create a end point to receive alerting event from grafana. Convert that alert event into current alerting structure. And pass the alert into node-agent socket.
+
+== Implementation:
+
+A new repository for Monitoring-Integration will be created.
+
+Following are the implementation details of how this functionality will be achieved.
+
+* Create a new file called alert_handler.py in monitoring-integration.
+
+* Create a new class called AlertHandler in alert_handler.py
+
+* Run AlertHandler as seperate gevent from monitoring-integration manager.
+
+* Create a function called ‘to_dict’ in AlertHandler to convert the received alert into dictionary based on current tendrl alert structure.
+
+* Create a new message object using alert dictionary as metadata.
+
+* Pass the message object into message socket using Event class.
+
+* Pass the created Event to the node-agent socket.
+
+* Notify the respective user about the alert.
+
+Webhook Receiver to receive the alerting events from grafana:
+Psuedo code:
+
+webhook_receiver():
+
+    socket.host =  localhost
+    socket.pot  =  1000008
+    while True:
+         conn,adress = socket.accept()
+         request = conn.recv(1024)
+         notify_alert_handler(request)
+
+=== Assignee(s):
+
+
+@GowthamShanmugam
+@rishubhjain
+
+
+=== Work Items:
+
+https://github.com/Tendrl/specifications/issues/169
+
+
+== Dependencies:
+
+None
+
+
+== Testing:
+
+Check all grafana and sds alerts are stored and notified to the users correctly.
+
+
+== Documentation impact:
+
+None
```

Comment:
Could someone from @Tendrl/tendrl-core verify that the impact is really none so that no changes in installation are needed here? I'm a little confused about this.